### DISCUSSION TITLE
[@types/node] stream/consumers json() incorrect return typing

### DISCUSSION THREAD

**Discussion by acommodari** [DefinitelyTyped/DefinitelyTyped#61955] [Issues with a @types package] [answered]
`import { json } from node:stream/consumers;` has a return type of `Promise<unknown>` in v16, v17, and v18  of `@types/node`.

If you look at the actual source [code](https://github.com/nodejs/node/blob/main/lib/stream/consumers.js#L74) node says it should have a type of `Promise<any>`. This also makes sense if you look at the implementation because it's just calling the `text()` function and wrapping it in `JSON.parse()` which also has a return type of `Promise<any>`.

I know that this change will definitely help me resolve some type errors that I think are unwarranted without me having to jump through overly complex type guards. 😄 

Thanks in advance!

[a comment by an automated account is not shown]

**Comment by acommodari** [comment]
Just checking in on this discussion 😄

**Comment by SimonSchick** [comment]
Please just use a type assertion, the usage of `any` is generally considered unsafe.

**Comment by LinusU** [comment, marked as answer]
I personally would very much prefer `unknown`, since `any` is very easy to mess up. I think that the only reason that `JSON.parse` is typed as `any`, is because `unknown` didn't exists at the time those types were created.

We have had a similar discussion about this in node-fetch: https://github.com/node-fetch/node-fetch/issues/1262

**Reply by LinusU** [reply]
Some more discussion: https://github.com/microsoft/TypeScript/issues/26188
